service/src/main/java/com/softserve/borda/services/impl/BoardServiceImpl.java

```diff
@@ -70,4 +74,17 @@ public boolean deleteBoardById(Long id) {
                 .map(UserBoardRelation::getBoard)
                 .collect(Collectors.toList());
     }
+
+    @Override
+    public Board addTagToBoard(Long boardId, Long tagId) {
+        Board board = getBoardById(boardId);
+        Tag tag = tagService.getTagById(tagId);
+        board.getTags().add(tag);
+        return boardRepository.save(board);
+    }
+
+    @Override
+    public List<Tag> getAllTagsByBoardId(Long boardId) {
+        return getBoardById(boardId).getTags();
```

Comment:
Do we really need all data about a board in this case?
I mean, possibly it is better to fetch data about tags from TagRepository?

Reply:
Right now we have unidirectional mapping between Board and Tag. Board has list of Tags, but Tags have no link to Board. So I am not sure how I should do it then. Is it just better to use bidirectional mapping? Then I will be able to rework it from TagRepository.